source/server/connection_handler_impl.cc

```diff
@@ -401,6 +445,38 @@ void ConnectionHandlerImpl::ActiveTcpListener::newConnection(
     return;
   }
 
+  // const auto filter_chain_impl = dynamic_cast<const Server::FilterChainImpl*>(filter_chain);
+  bool is_fake_filter_chain = filter_chain->isFakeFilterChain();
+  // bool is_fake_filter_chain = filter_chain_impl->isFakeFilterChain();
+  if (is_fake_filter_chain) {
+    ENVOY_LOG(debug, "found a filter chain placeholder, start rebuilding request");
+    const auto& worker_name = parent_.dispatcher_.name();
+    const auto& listener_name = config_->name();
+    ENVOY_LOG(debug, "info: worker_name: {}, listener_name: {}", worker_name, listener_name);
+
+    auto& listener = dynamic_cast<ListenerImpl&>(*config_);
+    auto& server_dispatcher = listener.dispatcher();
+    // const auto& filter_chain_message = filter_chain_impl->getFilterChainMessage();
+    const auto& filter_chain_message = filter_chain->getFilterChainMessage();
+
+    parent_.sockets_using_filter_chain_[filter_chain_message][listener_name].push_back(
+        std::make_pair(std::move(socket), dynamic_metadata));
```

Comment:
2. This might work. If it doesn't: emplace_back(std::move(socket), dynamic_metadata);

Reply:
Yes, updated in the latest commit.